### DISCUSSION TITLE
Invalid wav file

### DISCUSSION THREAD

**Discussion by eltoob** [deepgram/community#832] [General help] [answered]
### Which Deepgram product feature are you using?

Deepgram API - STT Pre-recorded

### Details

Hi,
I have an audio file (.wav) that seems to return 400 when I try to transcribe it (both from the playground and from the API)
I try to cut this audio in half using ffmpeg and it goes fine.
How can I debug the issue on why I get an error back from deepgram

I currently get that error: 
err_code:"Bad Request"
err_msg:"Bad Request: Invalid data received."
request_id:"09d36336-9135-406c-8fde-3004acb024d5"

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jjmaldonis** [comment, marked as answer]
Hey @eltoob, the debugging you've already done implies the audio data is corrupted part way through the file.

If you receive an "Invalid data received" error, you can use ffmpeg to convert the audio to a different format. When ffmpeg converts the audio, it does its best to read the original audio file. If there is a corrupt portion of the audio, ffmpeg will do its best to work around that corrupt portion and continue to convert the full audio. When ffmpeg completes the conversion, you will have a new audio file that should be the same as the original, but it won't be corrupt. If ffmpeg could not read the original audio file correctly, the before/after files will be different.

You can add the `-v trace` option when converting the file from one format to another to get more info about the file's contents as ffmpeg decodes it. This often reveals the issue with the audio data.
